org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/mapping/AbstractDoEntityMapper.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2010-2021 BSI Business Systems Integration AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     BSI Business Systems Integration AG - initial API and implementation
+ */
+package org.eclipse.scout.rt.dataobject.mapping;
+
+import static org.eclipse.scout.rt.platform.util.Assertions.assertNotNull;
+
+import org.eclipse.scout.rt.dataobject.IDoEntity;
+import org.eclipse.scout.rt.platform.BEANS;
+import org.eclipse.scout.rt.platform.Platform;
+
+/**
+ * {@link DoEntityMappings} container class allowing to map from a DO entity to a peer and vice versa.
+ * <p>
+ * Implementation are usually application scoped so that the mappings are only collected once.
+ * <p>
+ * Definition example:
+ *
+ * <pre>
+ * &#64;ApplicationScoped
+ * public static class LoremFormMapper extends AbstractDoEntityMapper<LoremDo, LoremForm> {
+ *
+ *   &#64;Override
+ *   protected void initMappings(DoEntityMappings<LoremDo, LoremForm> mappings) {
+ *     mappings
+ *         .with(LoremDo::key, LoremForm::getKey, LoremForm::setKey)
+ *         .withHolder(LoremDo::name, LoremForm::getNameField);
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * Usage Example within an {@link org.eclipse.scout.rt.client.ui.form.AbstractFormHandler}
+ * <ul>
+ * <li>execLoad: <code>BEANS.get(LoremFormMapper.class).fromDo(data, (LoremForm) form);</code>
+ * <li>execStore: <code>BEANS.get(LoremFormMapper.class).&#116;oDo((LoremForm)form, data);</code>
+ * </ul>
+ */
+public abstract class AbstractDoEntityMapper<DO_ENTITY extends IDoEntity, PEER> {
+
+  protected DoEntityMappings<DO_ENTITY, PEER> m_mappings = new DoEntityMappings<>();
+
+  public AbstractDoEntityMapper() {
+    initMappings(m_mappings);
```

Comment:
-> could call reset() here
-> m_mappings = new DoEntityMappings<>() above would not be necessary